From 1f35b2224a20b9df9de1d1a904d201764d86cee7 Mon Sep 17 00:00:00 2001
Message-Id: <1f35b2224a20b9df9de1d1a904d201764d86cee7.1413837856.git.chang-joon.lee@intel.com>
In-Reply-To: <006497b7323eca32c2f489a1a9f662743c0d16d2.1413837855.git.chang-joon.lee@intel.com>
References: <006497b7323eca32c2f489a1a9f662743c0d16d2.1413837855.git.chang-joon.lee@intel.com>
From: Oscar Mateo <oscar.mateo@intel.com>
Date: Tue, 3 Dec 2013 16:00:31 +0000
Subject: [PATCH 016/312] REVERTME [VPG]: drm: removing DRM Master check on
 the DRM ioctls

Some ioctls (for example for mode setting) shall be allowed only for
the DRM master (whosever opens the DRM device file first will become
the DRM master). On Android side, SurfaceFlinger process is expected
to act like a DRM Master but, currently, DRM master check is failing
on the ioctls made by SurfaceFlinger process. This is because some
other service/process is opening the DRM device file & becoming a
master. For now, master check is not done.

Also for Android, currently all the Users are by default being marked
as authenticated. There is no X-Server like thing over here, which is
there in Linux desktop systems, which can act like a DRM master &
authenticate the clients. So currently all the DRM users/clients are
by default being marked as authenticated.

Originally, this patch was:
commit b72089dba5211f25a9e2450b3854c06bfa17a5ec
Author: akashgoe <akash.goel@intel.com>
Date:   Wed Sep 18 09:03:08 2013 +0530

    drm/i915: Adding Root permission check on the DRM/i915 ioctls

Issue: APDEV-983
Change-Id: I5cbfbbd6b7f60c1b77211f8666854f294ac72907
Signed-off-by: Oscar Mateo <oscar.mateo@intel.com>
Signed-off-by: A.Sunil Kamath <sunil.kamath@intel.com>
(cherry picked from commit 468cac97ad3fa5f81c89bd940858d5853c2bcb4e)

Signed-off-by: Jon Bloomfield <jon.bloomfield@intel.com>
---
 drivers/gpu/drm/drm_drv.c |    6 ++++++
 1 file changed, 6 insertions(+)

diff --git a/drivers/gpu/drm/drm_drv.c b/drivers/gpu/drm/drm_drv.c
index 8218078..ba09c26 100644
--- a/drivers/gpu/drm/drm_drv.c
+++ b/drivers/gpu/drm/drm_drv.c
@@ -306,10 +306,16 @@ static int drm_ioctl_permit(u32 flags, struct drm_file *file_priv)
 		     !file_priv->authenticated))
 		return -EACCES;
 
+/* FIXME: On Android side, SurfaceFlinger process is expected to act like a
+    DRM Master, but currently DRM master check is failing on the ioctls made by
+    SurfaceFlinger process. This is because some other service/process is opening the
+    DRM device file & becoming a master */
+#if 0
 	/* MASTER is only for master or control clients */
 	if (unlikely((flags & DRM_MASTER) && !file_priv->is_master &&
 		     !drm_is_control_client(file_priv)))
 		return -EACCES;
+#endif
 
 	/* Control clients must be explicitly allowed */
 	if (unlikely(!(flags & DRM_CONTROL_ALLOW) &&
-- 
1.7.9.5

